src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -270,6 +319,16 @@ private void InstallButton_Clicked(object sender, RoutedEventArgs e)
         {
             if (InstallButtonClicked != null)
             {
+                // If the user didnt used the enter key, the version range can be different
+                VersionRange userRequestedVersionRange = null;
+                var isUserInputValidNuGetVersionRange = VersionRange.TryParse(_versions.Text, out userRequestedVersionRange);
+
+                if (isUserInputValidNuGetVersionRange && PackageDetailControlModel.SelectedVersion.Range != userRequestedVersionRange)
+                {
+                    IEnumerable<NuGetVersion> versions = DetailModel.Versions.Where(v => v != null).Select(v => v.Version);
+                    NuGetVersion rangeBestVersion = userRequestedVersionRange.FindBestMatch(versions);
+                    PackageDetailControlModel.SelectedVersion = new DisplayVersion(userRequestedVersionRange, rangeBestVersion, additionalInfo: null);
```

Comment:
Why isn't this already handled somewhere before clicking the Install button? Can you help me understand this scenario?

If user changes combobox, events fire which update `SelectedVersion`, right?

Reply:
You're right, the problem was with the range not being correctly populated, but I can fix it when we update SelectedVersion